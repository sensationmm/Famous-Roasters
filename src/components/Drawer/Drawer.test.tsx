import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'

import { Drawer } from '.'

const intersectionObserverMock = function () {
  return {
    observe: jest.fn(),
    disconnect: jest.fn(),
  }
}
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
window.IntersectionObserver = intersectionObserverMock

describe('Drawer component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <Drawer trigger={<button>CLick me</button>} title="My drawer" body={<div>My content</div>} />,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Can be triggered and dismissed', async () => {
    render(<Drawer trigger={<button>CLick me</button>} title="My drawer" body={<div>My content</div>} />)
    const buttonTrigger = await screen.findByTestId('drawer-trigger')
    expect(buttonTrigger).toBeInTheDocument()
    fireEvent.click(buttonTrigger)
    const buttonDismiss = await screen.findByTestId('drawer-dismiss')
    expect(buttonDismiss).toBeInTheDocument()
    fireEvent.click(buttonDismiss)
  })
})
