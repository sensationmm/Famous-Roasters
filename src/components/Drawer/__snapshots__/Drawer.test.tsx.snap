// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component Renders correctly 1`] = `
<div>
  <div>
    <button
      data-testid="drawer-trigger"
    >
      CLick me
    </button>
  </div>
</div>
`;
